hosting/src/views/Event/Event.ts

```diff
@@ -3,10 +3,15 @@ import { Vue } from 'vue-class-component'
 import { useMeta } from 'vue-meta'
 import { useRoute } from 'vue-router'
 
+import { EventUi } from '@/models/event'
 import { Route } from '@/router/route-decorator'
 
 @Route({ name: 'Event', path: '/events/:slug' })
 export default class Event extends Vue {
   route = useRoute()
   meta = useMeta(computed(() => ({ title: `Event ${this.route.params.slug as string}` })))
+
+  get event(): EventUi {
+    return this.$store.getters.eventBySlug(this.route.params.slug)
```

Comment:
Ah ha! I have something to show you about decoupling route params!

Reply:
I've implemented this in `Event.ts` (at least... the most basic case):
```ts
@Route({ name: 'Event', path: '/events/:slug', props: true })
export default class Event extends Vue.with(class {
  slug!: string
}) {
   ...

  get event(): EventUi | undefined {
    return this.store.getters.eventBySlug(this.slug) // no need to access $route
  }
}
```

The main idea is that Components should not be coupled to `route/$route`; only to `props`. This means multiple routes can use the same Component, even if their params are different. A `vue-router` route definition can provide what is essentially a [mapping function](https://router.vuejs.org/guide/essentials/passing-props.html#function-mode) that converts its params to the Component's props.

If the names & types of params & props are identical, there is also a simpler [boolean switch to pass-thru params](https://router.vuejs.org/guide/essentials/passing-props.html#boolean-mode) (which is all I needed so far).

I will need to improve the type signature for `props` in my decorator to support the function mode.

Reply:
DONE